webapp/src/ts/modules/contacts/contacts-content.component.ts

```diff
@@ -153,4 +199,182 @@ export class ContactsContentComponent implements OnInit, OnDestroy {
+    await this.setSettings();
+
+    this.globalActions.setRightActionBar({
+      relevantForms: [], // This disables the "New Action" button until forms load
+      sendTo: this.selectedContact?.type?.person ? this.selectedContact?.doc : '',
+      canDelete: this.canDeleteContact,
+      canEdit: this.sessionService.isAdmin() || this.userSettings?.facility_id !== this.selectedContact?.doc?._id,
+      openContactMutedModal: (form) => this.openContactMutedModal(form),
+      openSendMessageModal: (sendTo) => this.openSendMessageModal(sendTo)
+    });
+
+    this.subscribeToAllContactXmlForms();
+    this.subscribeToSelectedContactXmlForms();
+  }
+
+  private setUserSettings() {
+    if (this.userSettings) {
+      return;
+    }
+    return this.userSettingsService
+      .get()
+      .then(userSettings => this.userSettings = userSettings)
+      .catch(e => console.error('Error fetching user settings', e));
+  }
+
+  private setSettings() {
+    if (this.settings) {
+      return;
+    }
+    return this.settingsService
+      .get()
+      .then(settings => this.settings = settings)
+      .catch(e => console.error('Error fetching settings', e));
+  }
+
+  private async setChildTypesBySelectedContact() {
+    if (!this.selectedContact) {
+      this.childTypesBySelectedContact = [];
+      return;
+    }
+
+    if (!this.selectedContact.type) {
+      const type = this.selectedContact.doc?.contact_type || this.selectedContact.doc?.type;
+      console.error(`Unknown contact type "${type}" for contact "${this.selectedContact.doc?._id}"`);
+      this.childTypesBySelectedContact = [];
+      return;
+    }
+
+    return this.contactTypesService
+      .getChildren(this.selectedContact.type.id)
+      .then(childTypes => this.childTypesBySelectedContact = childTypes)
+      .catch(e => console.error('Error fetching contact child types', e));
+  }
+
+  private subscribeToAllContactXmlForms() {
+    if (this.subscriptionAllContactForms) {
+      this.subscriptionAllContactForms.unsubscribe();
+    }
+
+    this.subscriptionAllContactForms = this.xmlFormsService.subscribe(
+      'ContactForms',
+      { contactForms: true },
+      (error, forms) => {
+        if (error) {
+          console.error('Error fetching allowed contact forms', error);
+          return;
+        }
+
+        const allowedChildTypes = this.filterAllowedChildType(forms, this.childTypesBySelectedContact);
+        this.globalActions.updateRightActionBar({
+          childTypes: this.getModelsFromChildTypes(allowedChildTypes)
+        });
+      }
+    );
+    this.subscription.add(this.subscriptionAllContactForms);
+  }
+
+  private subscribeToSelectedContactXmlForms() {
+    if (!this.selectedContact) {
+      return;
+    }
+
+    if (this.subscriptionSelectedContactForms) {
+      this.subscriptionSelectedContactForms.unsubscribe();
+    }
+
+    this.subscriptionSelectedContactForms = this.xmlFormsService.subscribe(
+      'ContactList',
```

Comment:
Can we use a more intuitive name? Like "selectedContactForms" or something like that?
"ContactList" makes no sense.

Reply:
Absolutely! Fixed